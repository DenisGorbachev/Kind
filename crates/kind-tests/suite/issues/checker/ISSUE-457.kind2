#derive[match]
record Pair (a: Type) (b: Type) {
    fst : a
    snd : b
}

A (n: Pair U60 U60) : U60
A n =
    match Pair n {
        new => ?
    }


B (n: Pair U60 U60) : U60
B n =
    open Pair n
    ?

C (n: Pair U60 U60) : U60
C n =
    let Pair.new fst snd = n
    ?


