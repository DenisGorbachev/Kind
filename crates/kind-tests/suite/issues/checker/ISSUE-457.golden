   INFO  Inspection

      * Hole: Data.U60 

      * Context
      * n     : (Data.Pair Data.U60 Data.U60) 
      * n.fst : Data.U60 
      * n.snd : Data.U60 

      /--[suite/issues/checker/ISSUE-457.kind2:10:16]
      |
    9 |        match Data.Pair n {
   10 |            new => ?
      |                   v
      |                   \Here!
   11 |        }


   INFO  Inspection

      * Hole: Data.U60 

      * Context
      * n     : (Data.Pair Data.U60 Data.U60) 
      * n.fst : Data.U60 
      * n.snd : Data.U60 

      /--[suite/issues/checker/ISSUE-457.kind2:17:5]
      |
   16 |        open Data.Pair n
   17 |        ?
      |        v
      |        \Here!
   18 |    


   INFO  Inspection

      * Hole: Data.U60 

      * Context
      * n   : (Data.Pair Data.U60 Data.U60) 
      * fst : Data.U60 
      * snd : Data.U60 

      /--[suite/issues/checker/ISSUE-457.kind2:22:5]
      |
   21 |        let Data.Pair.new fst snd = n
   22 |        ?
      |        v
      |        \Here!
   23 |    


